### PR TITLE
Adding basic test for rendering upstream helm hooks to base

### PR DESCRIPTION
**Pull request by jala-dx**

### CODE REVIEW

File: pkg/tests/base/render_test.go [hunk 1]
```diff
@@ -72,10 +73,11 @@ func TestRenderUpstream(t *testing.T) {
 
 		test.Upstream.Files = upstreamFilesFromDir(t, filepath.Join(path, "upstream"))
 
-		test.WantBase = baseFromDir(t, filepath.Join(path, "base"))
-
-		// TODO: helm bases
-
+		test.WantBase = baseFromDir(t, filepath.Join(path, "base"), false)
+		helmpath := filepath.Join(path, "basehelm")
+		if _, err := os.Stat(helmpath); !os.IsNotExist(err) {
```
Comment (sgalsaleh) [inline pkg/tests/base/render_test.go:78]:
```suggestion
		if _, err := os.Stat(helmpath); err == nil {
```

File: pkg/tests/base/render_test.go [hunk 2]
```diff
@@ -86,24 +88,41 @@ func TestRenderUpstream(t *testing.T) {
...
+						if len(gotBase.Files) > idx && gotBase.Files[idx].Path == tt.WantBase.Files[idx].Path {
+							t.Log("FILE", tt.WantBase.Files[idx].Path)
+							t.Log(diffString(string(gotBase.Files[idx].Content), string(tt.WantBase.Files[idx].Content)))
+						}
+					}
+					t.FailNow()
+				}
+			}
+			if len(tt.WantHelmBase.Files) > 0 {
+				gotHelmBase := gotHelmBases[0]
```
Comment (sgalsaleh) [inline pkg/tests/base/render_test.go:110]: why only validate one base? why not have `tt.WantHelmBase` be an array and validate all bases?

Comment (jala-dx) [reply pkg/tests/base/render_test.go:110]: I need to think a way to add a testcase to handle multiple charts. will add a TODO.

**Review by sgalsaleh — APPROVED** [review]

File: pkg/tests/base/render_test.go [hunk 3]
```diff
@@ -86,24 +88,42 @@ func TestRenderUpstream(t *testing.T) {
...
+					for idx := range tt.WantHelmBase.Files {
+						if len(gotBase.Files) > idx && gotBase.Files[idx].Path == tt.WantBase.Files[idx].Path {
+							t.Log("FILE", tt.WantHelmBase.Files[idx].Path)
+							t.Log("FILE", gotHelmBases[0].Files[idx].Path)
+							t.Log(diffString(string(gotHelmBases[0].Files[idx].Content), string(tt.WantHelmBase.Files[idx].Content)))
+						}
 					}
+					t.FailNow()
 				}
-				t.FailNow()
```
Comment (divolgin) [inline pkg/tests/base/render_test.go:103]: This test will pass but should fail if first check fails and the second one passes.

File: pkg/tests/base/render_test.go [hunk 4]
```diff
@@ -86,24 +88,42 @@ func TestRenderUpstream(t *testing.T) {
...
+					t.FailNow()
+				}
+			}
+			if len(tt.WantHelmBase.Files) > 0 {
+				// TODO handle multiple charts
+				gotHelmBase := gotHelmBases[0]
+				if !assert.IsEqual(tt.WantHelmBase.Files, gotHelmBase.Files) {
+					t.Log(diffJSON(gotHelmBases, tt.WantHelmBase))
+				}
+				if !assert.IsEqual(tt.WantBase.Files, gotBase.Files) {
```
Comment (sgalsaleh) [inline pkg/tests/base/render_test.go:115]: should this be `WantHelmBase` and `gotHelmBase`?

File: pkg/tests/base/render_test.go (hunk 2 above)
Comment (divolgin) [reply pkg/tests/base/render_test.go:111]: This should assert that `len(gotHelmBases) == 1`

File: pkg/tests/base/render_test.go [hunk 5]
```diff
@@ -86,24 +88,42 @@ func TestRenderUpstream(t *testing.T) {
...
+					for idx := range tt.WantBase.Files {
+						if len(gotBase.Files) > idx && gotBase.Files[idx].Path == tt.WantBase.Files[idx].Path {
+							t.Log("FILE", tt.WantBase.Files[idx].Path)
+							t.Log(diffString(string(gotBase.Files[idx].Content), string(tt.WantBase.Files[idx].Content)))
+						}
+					}
+					t.FailNow()
+				}
+			}
+			if len(tt.WantHelmBase.Files) > 0 {
```
Comment (divolgin) [inline pkg/tests/base/render_test.go:109]: This check shouldn't be needed.  Number of items in `gotHelmBases` should be checked instead

**Review by sgalsaleh — APPROVED** [review]

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**